kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java

```diff
@@ -460,41 +477,52 @@ public static Config fromKubeconfig(String kubeconfigContents) throws IOExceptio
   public static Config fromKubeconfig(String context, String kubeconfigContents, String kubeconfigPath) {
     // we allow passing context along here, since downstream accepts it
     Config config = new Config();
-    Config.loadFromKubeconfig(config, context, kubeconfigContents, kubeconfigPath);
+    loadFromKubeconfig(config, context, kubeconfigContents, kubeconfigPath);
     return config;
   }
 
   private static boolean tryKubeConfig(Config config, String context) {
     LOGGER.debug("Trying to configure client from Kubernetes config...");
-    if (Utils.getSystemPropertyOrEnvVar(KUBERNETES_AUTH_TRYKUBECONFIG_SYSTEM_PROPERTY, true)) {
-      String fileName = Utils.getSystemPropertyOrEnvVar(KUBERNETES_KUBECONFIG_FILE, new File(getHomeDir(), ".kube" + File.separator + "config").toString());
+    if (!Utils.getSystemPropertyOrEnvVar(KUBERNETES_AUTH_TRYKUBECONFIG_SYSTEM_PROPERTY, true)) {
+      return false;
+    }
+    File kubeConfigFile = new File(getKubeconfigFilename());
+    if (!kubeConfigFile.isFile()) {
+      LOGGER.debug("Did not find Kubernetes config at: ["+kubeConfigFile.getPath()+"]. Ignoring.");
```

Comment:
nit, Could you please use format specifiers?

Reply:
> nit, Could you please use format specifiers?

you mean the slf4j facility, right?